### DISCUSSION TITLE
Self-hosted supabase - RLS on _realtime and _analytics schema

### DISCUSSION THREAD

**Discussion by fredericauvert** [supabase/supabase#18566] [Questions]
Hello,

I'm trying to understand and work with a selft hosted supabase. All is working fine with my RLS configuration on public tables.

When I watch into the configuration, under Authentication -> Policies, RLS is correctly activated for tables in public schema but RLS is not activated for _realtime and _analytics schema.

Is it normal and safe ?
Should I have to enable RLS for tables under those two shemas ?

**Comment by GaryAustin1** [comment]
In general you should only put RLS on tables in public, storage or your own custom schemas.  Do not change schemas you do not own.  RLS is also only useful if they are accessible to the API with an API setting and roles that obey RLS have been granted use of them.

**Reply by fredericauvert** [reply]
Thank you for your answer.
I'm understanding that and agree.
What is not clear for me is both schemas _realtime and _analytics seems to be reachable with the anon key and have no RLS enable then someone could play with tables in those schemas.
I will make some tests when I will have time to understand better if it's really reachable and what the risk to leave them without RLS.

**Reply by GaryAustin1** [reply]
Unless they are in the API settings they can't be reached by the API.
Note these tables seem to be only for selfhosted and are not on hosted instances.   I guess it is possible this is an oversite.
